### PR TITLE
Development/Vortex: Initial Commit

### PR DESCRIPTION
**Pull request by Kreyren**
Resolves : https://github.com/PhoenicisOrg/scripts/issues/903

HELP_WANTED: Results in `error=2 no file or directory` after the installation
image

Signed-off-by: Jacob Hrbek <werifgx@gmail.com>

### CODE REVIEW

**Comment by plata** [comment]
There's no script.js file?

**Comment by Kreyren** [comment]
> There's no script.js file?

wierd, fixing.

**Comment by Kreyren** [comment]
Updated OP with error.

**Comment by Kreyren** [comment]
Same(?) issue present if vortex is installed using `Online Installer` 
image

**Comment by Kreyren** [comment]
@ImperatorS79 I require your devine powers :pray:

**Comment by plata** [comment]
Seems like the relevant part of the error message is cut off in your screenshot.

**Comment by ImperatorS79** [comment]
If there is a wine executable in the directory pointed in your screenshot, you might be missing 32 bits deps.

**Comment by Kreyren** [comment]
```
java.lang.RuntimeException: java.io.IOException: Cannot run program "/home/kreyren/.Phoenicis/engines//wine/staging-linux-amd64/4.3/bin/wine" (in directory "/home/kreyren/.Phoenicis/containers/wineprefix/Vortex"): error=2, No such file or directory
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.linker.JavaAdapterServices.wrapThrowable(JavaAdapterServices.java:285)
	at jdk.nashorn.javaadapters.org_phoenicis_engines_Engine.run(Unknown Source)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$501$4947AAAZZA$\^eval\_.run(engines.wine.engine.object:178)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$500$219A$\^eval\_$cu1$restOf.regedit#open(engines.wine.plugins.regedit:12)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$486$287$\^eval\_$cu1$restOf.luna(engines.wine.verbs.luna:29)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$453$493$\^eval\_$cu1$restOf.go(engines.wine.quick_script.installer_script:72)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$388$425$\^eval\_.run(<eval>:12)
	at jdk.nashorn.javaadapters.org_phoenicis_scripts_Installer.run(Unknown Source)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$387$\^eval\_$cu1$restOf.:program(<eval>:39)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:655)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:513)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:527)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:456)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:413)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:409)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.eval(NashornScriptEngine.java:162)
	at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
	at org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:50)
	at org.phoenicis.scripts.nashorn.NashornScriptInterpreter.runScript(NashornScriptInterpreter.java:35)
	at org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$runScript$0(BackgroundScriptInterpreter.java:35)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.IOException: Cannot run program "/home/kreyren/.Phoenicis/engines//wine/staging-linux-amd64/4.3/bin/wine" (in directory "/home/kreyren/.Phoenicis/containers/wineprefix/Vortex"): error=2, No such file or directory
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$525$7170AAAZZA$\^eval\_$cu1$restOf.run(engines.wine.engine.implementation:272)
	... 22 more
Caused by: java.io.IOException: error=2, No such file or directory
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:340)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:271)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
	... 24 more
```

```console
kreyren@dreamon ~/GIT/compatibilators $ ~/.Phoenicis/engines/wine/staging-linux-amd64/4.3/bin/wine --version
wine-4.3 (Staging)
kreyren@dreamon ~/GIT/compatibilators $ ls ~/.Phoenicis/containers/wineprefix/Vortex
drive_c  phoenicis.cfg  wine.log
kreyren@dreamon ~/GIT/compatibilators $ ls ~/.Phoenicis/containers/wineprefix/Vortex/drive_c/
windows
```

**Comment by Kreyren** [comment]
> If there is a wine executable in the directory pointed in your screenshot, you might be missing 32 bits deps.

or elaborate 32-bit deps?

**Comment by ImperatorS79** [comment]
You are maybe missing the required dependencies required to run 32 bits programs (on a 64 bits OS)(libc, libstd, etc).

**Comment by ImperatorS79** [comment]
@Kreyren Could you update this ? Be sure to have libc6:i386 installed (or use the .deb from latest phoenicis master).

File: Applications/Development/Vortex/v0.17.10-WINE/script.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            // .url("https://www.nexusmods.com/Core/Libs/Common/Widgets/DownloadPopUp?id=159&game_id=2295&source=FileExpander") // For Vortex without custom install
+            // TODO: Add option for other url if required.
+            .url("https://github.com/Nexus-Mods/Vortex/releases/download/v0.17.10/vortex-setup-0.17.10.exe") // Source on https://github.com/Nexus-Mods/Vortex/releases
+            // TODO: Always download latest release!
+            .checksum("ab801475be427bb3c051ba7832704f482a6d1f4d") // Based on https://www.virustotal.com/#/file/7506b827c92e0052d66a4b191272764604538508ae75f3afe4747f69479d0aa8/details
+            .category("Development")
+            .wineVersion(LATEST_STAGING_VERSION)
+            .wineDistribution("staging")
+            .wineArchitecture("amd64")
+            .executable("vortex-setup-0.17.10.exe")
```
Comment (ImperatorS79) [inline Applications/Development/Vortex/v0.17.10-WINE/script.js]: This should be the final shortcut exe, not the installation ece

**Comment by Kreyren** [comment]
> @Kreyren Could you update this ? Be sure to have libc6:i386 installed (or use the .deb from latest phoenicis master).

How do i add these on phoenicis for 64-bit prefix? Shoudn't it use multilib by default?

File: Applications/Development/Vortex/v0.17.10-WINE/script.js (hunk 1 above)
Comment (Kreyren) [reply Applications/Development/Vortex/v0.17.10-WINE/script.js]: that is final afaik

**Comment by ImperatorS79** [comment]
Ah it is a 64 bits app, my bad ^^.

**Comment by plata** [comment]
Closed due to inactivity. Feel free to reopen if the app works.

**Pull request closed without merging**